includes/Plugin_Context.php

```diff
@@ -69,6 +69,26 @@ public function basename() {
 		return plugin_basename( $this->main_file );
 	}
 
+	/**
+	 * Returns the plugin main file.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @return string Plugin main file.
+	 */
+	public function main_file() {
+		if ( false === strpos( $this->main_file, '.php' ) ) {
```

Comment:
Finding actual main file here in this method is too late. May be it should be done in the constructor of `Plugin_Context`.

Reply:
ok, let's do it.